### ISSUE TITLE
Uncaught TypeError: _this.editor.languageMode.isLineCommentedAtBufferRow is not a function

### ISSUE THREAD

**Issue by JHK** [EddyLuten/line-breaker#5]
steps to reproduce:

1. Invoke the `Line Breaker: Break` feature

**Atom**: 1.22.0 x64
**Electron**: 1.6.15
**OS**: Fedora 26
**Thrown From**: [line-breaker](https://github.com/EddyLuten/line-breaker) package 0.4.0

### Stack Trace

Uncaught TypeError: _this.editor.languageMode.isLineCommentedAtBufferRow is not a function

```
At /home/julian/.atom/packages/line-breaker/lib/line-breaker.coffee:27

TypeError: _this.editor.languageMode.isLineCommentedAtBufferRow is not a function
    at /packages/line-breaker/lib/line-breaker.coffee:27:28
    at Array.every (native)
    at Object.break (/packages/line-breaker/lib/line-breaker.coffee:26:59)
    at /packages/line-breaker/lib/line-breaker.coffee:6:10)
    at CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app/src/command-registry.js:381:36)
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:621:22)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:412:28)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app/src/window-event-handler.js:100:42)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app/src/window-event-handler.js:3:65)
```

### Commands

```
     -0:13.7.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always)
     -0:09.2.0 editor:move-to-first-character-of-line (input.hidden-input)
     -0:09.0 editor:select-to-end-of-line (input.hidden-input)
     -0:08.8.0 intentions:highlight (input.hidden-input)
     -0:08.6.0 line-breaker:break (input.hidden-input)
```

### Non-Core Packages

```
antialiased 0.1.0 
atom-beautify 0.30.6 
atom-ctags 5.1.2 
atom-material-syntax 1.0.7 
atom-material-syntax-light 0.4.6 
atom-material-ui 2.0.6 
atom-notifier 0.5.1 
atom-xtext-package 1.0.3 
autocomplete-ctags 0.4.1 
browser-plus 0.0.98 
busy-signal 1.4.3 
custom-title 1.0.1 
file-icons 2.1.13 
geiger 2.1.0 
go-debug 1.5.2 
go-plus 5.6.0 
go-signature-statusbar 1.2.3 
goto-last-edit 0.3.3 
graphviz-preview-plus 1.5.1 
highlight-selected 0.13.1 
hyperclick 0.1.5 
intentions 1.1.5 
language-apache 1.7.0 
language-docker 1.1.8 
language-dot 1.7.0 
language-elixir 0.20.3 
language-elm 1.5.0 
language-erlang 3.2.0 
language-groovy 0.7.0 
language-latex 1.1.1 
language-nginx 0.8.0 
language-plantuml 0.2.0 
language-scala 1.1.9 
latex 0.49.0 
line-breaker 0.4.0 
linter 2.2.0 
linter-elixirc 1.7.0 
linter-elm-make 0.27.2 
linter-ruby 1.2.6 
linter-ui-default 1.6.10 
perltidy 4.0.0 
plantuml-preview 0.12.9 
pretty-json 1.6.4 
project-manager 3.3.5 
sort-lines 0.18.0 
symbol-gen 1.3.1 
test-jumper 0.6.2 
text-buffer 8.2.0 
Zen 0.18.0 
```

**Comment by DiThi** [comment]
Happening to me too, with Markdown files. It works well with CoffeeScript.

**Comment by Beefster09** [comment]
This package used to work, but this error happens every time now.

FIX IT!

**Comment by DiThi** [comment]
@Beefster09 I don't think it's a good idea to _demand_ free work. It would probably help to be more polite. Also, the author is not the only one that can fix it. I may give it a try when I need it.

**Comment by marcmoy** [comment]
I really like this package too so I tried to start debugging this issue. Wasn't able to find the issue but I noticed that the `Atom::workspace` object in this repo is expecting a `languageMode` property but I can't find that in Atom's public API docs: [https://atom.io/docs/api/search/v1.16.0?utf8=%E2%9C%93&q=languageMode](https://atom.io/docs/api/search/v1.16.0?utf8=%E2%9C%93&q=languageMode)

**Comment by crachon** [comment]
Couldn't you just add a default languageMode as a first workaround?

**Comment by charrismatic** [comment]
Having the same problem on Atom 1.26

Just playing around with some options I was able to get it sort of working again. 

For the sake of experimenting I tried bypassing the isLineCommentedAtBufferedRow line to see what happened
```coffee
        # Is every single line in the selection a comment?
        isComments = [selection.start.row..selection.end.row].every (row) =>
        # @editor.languageMode.isLineCommentedAtBufferRow(row)
        true
```

Then Atom was complaining that  `@toggleComments`  is not a function.  I replaced that function with the [functions definition](https://github.com/atom/atom/blob/v1.26.1/src/selection.js#L686) and using the same parameters as inputs and it kind of worked. 

```coffee
        beforeUncomment = firstLineLength()
        this.editor.toggleLineCommentsForBufferRows(selection.start.row, selection.end.row)
        #  @toggleComments(selection.start.row, selection.end.row)
        afterUncomment = firstLineLength()
```


The line break functions are working again but I had to always have at least one line before or after be not in the comment block, and the comments were not flipping correctly. Also it was inserting new lines between where there shouldn't be. 

Maybe this will be helpful to the next person.

**Comment by charrismatic** [comment]
ok I got it working again using with the source for the `isLineCommentedAtBufferRow` function. I couldnt find the coffeescript version in the official repo but I used a [this version from a mirror](https://git.d4ni.nl/daniwinter/atom/src/70b0ec7471dd74b80ec5a25a67b31f4ef10ea92b/src/language-mode.coffee ) to get it working again 

```coffee
  # Is every single line in the selection a comment?
    isComments = [selection.start.row..selection.end.row].every (row) =>
      return false unless 0 <= row <= @editor.getLastBufferRow()
      @editor.tokenizedBuffer.tokenizedLines[row]?.isComment()
```

**Comment by EddyLuten** [comment]
Update on this: the PR including the fix was tested and merged, released in 0.4.1.
